### PR TITLE
Add Spark Tensorflow Connector 

### PR DESCRIPTION
**Pull request by karthikvadla**

### CODE REVIEW

**Comment by karthikvadla** [comment]
@skavulya @joyeshmishra. Please review

File: spark/pom.xml [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+            </testResource>
+        </testResources>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.tensorflow</groupId>
+            <artifactId>tensorflow-hadoop</artifactId>
+            <version>1.0-01232017-SNAPSHOT</version>
+            <classifier>shaded-protobuf</classifier>
```
Comment (skavulya) [inline spark/pom.xml]: @jhseu We are hosting a snapshot of the tensorflow-hadoop jar on our maven repo https://tap.jfrog.io/tap/public-snapshots . Is there an alternative maven repo that we can use to host versioned tensorflow-hadoop jars, or should we change our build instructions to compile tensorflow-hadoop from source?

Comment (jhseu) [inline spark/pom.xml]: This is fine for now. I'll take a stab at uploading it elsewhere later.

Comment (skavulya) [inline spark/pom.xml]: Thanks!

**Comment by jhseu** [comment]
FYI: Code review might be delayed a few days due to deadlines this week

**Comment by skavulya** [comment]
No problem, @jhseu. In the meantime, we will update the licenses in the PR to match TensorFlow's.

We had also created a  [Spark package](https://spark-packages.org/package/tapanalyticstoolkit/spark-tensorflow-connector) which can be run as follows:
`$SPARK_HOME/bin/spark-shell --packages tapanalyticstoolkit:spark-tensorflow-connector:1.0.0-s_2.11`

Once the PR is merged, we can delete our package and you can re-register the package under the TensorFlow Ecosystem repo.

File: README.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ request.
   Marathon, deployed on top of Mesos.
 - [hadoop](hadoop) - TFRecord file InputFormat/OutputFormat for Hadoop MapReduce
   and Spark.
+- [spark](spark) - Spark Tensorflow Connector
```
Comment (jhseu) [inline README.md]: nit: TensorFlow

File: spark/LICENSE [hunk 3]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (jhseu) [inline spark/LICENSE]: Delete this file? Should be implied by the top-level license.

File: spark/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+[![Build Status](https://travis-ci.org/tapanalyticstoolkit/spark-tensorflow-connector.svg?branch=sbt)](https://travis-ci.org/tapanalyticstoolkit/spark-tensorflow-connector)
```
Comment (jhseu) [inline spark/README.md]: Delete? Since it points to another repoistory.

File: spark/build.sbt [hunk 5]
```diff
@@ -0,0 +1,93 @@
+name := "spark-tensorflow-connector"
+
+organization := "org.trustedanalytics"
```
Comment (jhseu) [inline spark/build.sbt]: Update?

File: spark/build.sbt [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+scalaVersion in Global := "2.11.8"
+
+spName := "tapanalyticstoolkit/spark-tensorflow-connector"
```
Comment (jhseu) [inline spark/build.sbt]: Let's rename this to `tensorflow/spark-tensorflow-connector`?

File: spark/build.sbt [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+// If you published your package to Maven Central for this release (must be done prior to spPublish)
+spIncludeMaven := false
+
+publishMavenStyle := true
+
+licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0"))
+
+pomExtra :=
+  <url>https://github.com/tapanalyticstoolkit/spark-tensorflow-connector</url>
```
Comment (jhseu) [inline spark/build.sbt]: Update git URLs?

File: spark/pom.xml [hunk 8]
```diff
@@ -0,0 +1,306 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.trustedanalytics</groupId>
```
Comment (jhseu) [inline spark/pom.xml]: org.tensorflow

File: spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala [hunk 9]
```diff
@@ -0,0 +1,67 @@
+/**
+ *  Copyright (c) 2016 Intel Corporation
```
Comment (jhseu) [inline spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala]: Can you change all of these copyright lines to say: `Copyright 2016 The TensorFlow Authors. All Rights Reserved.`?

(Note that regardless, Intel retains copyright. If Intel wants to, they can add themselves to this file list: https://github.com/tensorflow/tensorflow/blob/master/AUTHORS )

File: spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DataTypesConvertor.scala [hunk 10]
```diff
@@ -0,0 +1,34 @@
+package org.tensorflow.spark.datasources.tfrecords
```
Comment (jhseu) [inline spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DataTypesConvertor.scala]: Add a copyright/license comment here?

File: spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+ */
+class DefaultSource extends DataSourceRegister
+      with CreatableRelationProvider
+      with RelationProvider
+      with SchemaRelationProvider{
+
+  /**
+   * Short alias for spark-tensorflow data source.
+   */
+  override def shortName(): String = "tensorflow"
```
Comment (jhseu) [inline spark/src/main/scala/org/tensorflow/spark/datasources/tfrecords/DefaultSource.scala]: "tfrecords"

It's possible we'll have a new file format in the future and we'll distinguish it by name.

File: spark/README.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+val rdd = spark.sparkContext.parallelize(testRows)
+
+//Save DataFrame as TFRecords
+val df: DataFrame = spark.createDataFrame(rdd, schema)
+df.write.format("tensorflow").save(path)
+
+//Read TFRecords into DataFrame.
+//The DataFrame schema is inferred from the TFRecords if no custom schema is provided.
+val importedDf1: DataFrame = spark.read.format("tensorflow").load(path)
+importedDf1.show()
```
Comment (skavulya) [inline spark/README.md]: @karthikvadla Please update the example read and load statements to tfrecords as per Jonathan's comments

File: spark/build.sbt (hunk 5 above)
Comment (karthikvadla) [inline spark/build.sbt]: What you want organization name to be. org.tensorflow?

File: spark/build.sbt (hunk 6 above)
Comment (karthikvadla) [inline spark/build.sbt]: sure

**Comment by karthikvadla** [comment]
I have updated all the changes as suggested. Please let me know for any further changes to be made. @skavulya @jhseu

File: spark/build.sbt [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0"))
+
+pomExtra :=
+  <url>https://github.com/tensorflow/ecosystem</url>
+    <scm>
+      <url>git@github.com:tensorflow/ecosystem.git</url>
+      <connection>scm:git:git@github.com:tensorflow/ecosystem.git</connection>
+    </scm>
+    <developers>
+      <developer>
```
Comment (skavulya) [inline spark/build.sbt]: @jhseu should we delete the developer list from build.sbt as well?

Comment (jhseu) [inline spark/build.sbt]: Developer list works if you're ok with people potentially contacting you with questions. So it's up to you :)

Comment (skavulya) [inline spark/build.sbt]: Thanks! We are ok with people contacting us :)

**Pull request merged**